src/postgres-client/src/metrics.rs

```diff
@@ -19,44 +20,87 @@ pub struct PostgresClientMetrics {
 }
 
-impl PostgresClientMetrics {
+/// Metrics specific to [PostgresClient](crate::PostgresClient)'s internal
+/// workings.
+///
+/// Use [`for_purpose`](PostgresClientsMetrics::for_purpose) to create a
+/// [`PostgresClientMetrics`] when creating a client/connection pool.
+#[derive(Debug, Clone)]
+pub struct PostgresClientsMetrics {
+    pub(crate) connpool_size: UIntGaugeVec,
+    pub(crate) connpool_acquires: IntCounterVec,
+    pub(crate) connpool_acquire_seconds: CounterVec,
+    pub(crate) connpool_available: IntGaugeVec,
+    pub(crate) connpool_connections_created: CounterVec,
+    pub(crate) connpool_connection_errors: CounterVec,
+    pub(crate) connpool_ttl_reconnections: CounterVec,
+}
+
+impl PostgresClientsMetrics {
     /// Returns a new [PostgresClientMetrics] instance connected to the given registry.
     pub fn new(registry: &MetricsRegistry, prefix: &str) -> Self {
         Self {
             connpool_size: registry.register(metric!(
                 name: format!("{}_postgres_connpool_size", prefix),
                 help: "number of connections currently in pool",
+                var_labels: ["purpose"],
             )),
             connpool_acquires: registry.register(metric!(
                 name: format!("{}_postgres_connpool_acquires", prefix),
                 help: "times a connection has been acquired from pool",
+                var_labels: ["purpose"],
             )),
             connpool_acquire_seconds: registry.register(metric!(
                 name: format!("{}_postgres_connpool_acquire_seconds", prefix),
                 help: "time spent acquiring connections from pool",
+                var_labels: ["purpose"],
             )),
             connpool_available: registry.register(metric!(
                 name: format!("{}_postgres_connpool_available", prefix),
                 help: "available connections in the pool",
+                var_labels: ["purpose"],
             )),
             connpool_connections_created: registry.register(metric!(
                 name: format!("{}_postgres_connpool_connections_created", prefix),
                 help: "times a connection was created",
+                var_labels: ["purpose"],
             )),
             connpool_connection_errors: registry.register(metric!(
                 name: format!("{}_postgres_connpool_connection_errors", prefix),
                 help: "number of errors when establishing a new connection",
+                var_labels: ["purpose"],
             )),
             connpool_ttl_reconnections: registry.register(metric!(
                 name: format!("{}_postgres_connpool_ttl_reconnections", prefix),
                 help: "times a connection was recycled due to ttl",
+                var_labels: ["purpose"],
             )),
         }
     }
+
+    /// Creates a new [`PostgresClientMetrics`] with metrics scoped to the given
+    /// `purpose`.
+    pub fn for_purpose(&self, purpose: &str) -> PostgresClientMetrics {
```

Comment:
@danhhz I added the `purpose` label so that I can have metrics scoped to timelines, but it does mean that the postgres/pool metrics for persist now also have this label. I think the dashboards should continue working, because we would ignore the label, but still wanted to flag

Reply:
Seems fine, but tbh I'm a bit skeptical. Metrics definitely have costs, do we really think the granularity of breaking this out by timeline is worth it?

Reply:
Unclear to me! I did it because it was low enough effort and I had time during thanksgiving when US folks were out. But I'm very happy to just drop this commit!

Context: we currently have only one timeline (epoch millis). We will probably add at least a second timeline for catalog state. But then again, we will likely evolve the `TimestampOracle` API to allow `read_ts` calls that get timestamps for multiple timelines as one CRDB query, to be more efficient. In which latter case we wouldn't be able to separate metrics out by timeline.

Which means I should probably drop this commit, eh? 😅